src/arkode/arkode_sprkstep_io.c

```diff
@@ -0,0 +1,478 @@
+/*---------------------------------------------------------------
+ * Programmer(s): Cody J. Balos @ LLNL
+ *---------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2022, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ *---------------------------------------------------------------
+ * This is the implementation file for the optional input and
+ * output functions for the ARKODE SPRKStep time stepper module.
+ *
+ * NOTE: many functions currently in arkode_io.c will move here,
+ * with slightly different names.  The code transition will be
+ * minimal, but the documentation changes will be significant.
+ *--------------------------------------------------------------*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <sundials/sundials_math.h>
+#include <sundials/sundials_types.h>
+
+#include "arkode/arkode.h"
+#include "arkode/arkode_sprk.h"
+#include "arkode_sprkstep_impl.h"
+
+/*===============================================================
+  SPRKStep Optional input functions (wrappers for generic ARKODE
+  utility routines).  All are documented in arkode_io.c.
+  ===============================================================*/
+int SPRKStepSetInterpolantDegree(void* arkode_mem, int degree)
+{
+  if (degree < 0) { degree = ARK_INTERP_MAX_DEGREE; }
+  return (arkSetInterpolantDegree(arkode_mem, degree));
+}
+
+int SPRKStepSetInterpolantType(void* arkode_mem, int itype)
+{
+  return (arkSetInterpolantType(arkode_mem, itype));
+}
+
+int SPRKStepSetErrHandlerFn(void* arkode_mem, ARKErrHandlerFn ehfun, void* eh_data)
+{
+  return (arkSetErrHandlerFn(arkode_mem, ehfun, eh_data));
+}
+
+int SPRKStepSetErrFile(void* arkode_mem, FILE* errfp)
+{
+  return (arkSetErrFile(arkode_mem, errfp));
+}
+
+int SPRKStepSetMaxNumSteps(void* arkode_mem, long int mxsteps)
+{
+  return (arkSetMaxNumSteps(arkode_mem, mxsteps));
+}
+
+int SPRKStepSetMaxHnilWarns(void* arkode_mem, int mxhnil)
+{
+  return (arkSetMaxHnilWarns(arkode_mem, mxhnil));
+}
+
```

Comment:
I recommend that this be removed from the public UI until time adaptivity is enabled.